File: frontend/src/types/portfolio.ts
export type PlatformType = 'youtube' | 'tiktok' | 'instagram' | 'naver'

export interface ChannelStat {
  id: number
  platform: PlatformType
  channelName: string
  subscribers: number
  totalViews: number
  avgViews: number
  engagementRate: number
  profileImageUrl: string
}

export interface ShowcaseContent {
  id: number
  title: string
  thumbnailUrl: string
  platform: PlatformType
  viewCount: number
  likeCount: number
  publishedAt: string
  videoUrl: string
  order: number
}

export interface BrandCollaboration {
  id: number
  brandName: string
  brandLogoUrl: string
  campaignTitle: string
  platform: PlatformType
  deliverables: string
  completedAt: string
  resultSummary: string
}

export type PortfolioTheme = 'minimal' | 'creative' | 'professional' | 'bold'

export interface PortfolioProfile {
  displayName: string
  bio: string
  profileImageUrl: string
  categories: string[]
  contactEmail: string
  website: string
}

export interface Portfolio {
  id: number
  profile: PortfolioProfile
  channelStats: ChannelStat[]
  showcaseContents: ShowcaseContent[]
  brandCollaborations: BrandCollaboration[]
  theme: PortfolioTheme
  slug: string
  isPublic: boolean
  publicUrl: string
  createdAt: string
  updatedAt: string
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/portfoliobuilder/Portfolio.kt
package com.ongo.domain.portfoliobuilder

import java.time.LocalDateTime

data class Portfolio(
    val id: Long? = null,
    val userId: Long,
    val title: String,
    val description: String = "",
    val template: String = "MINIMAL",
    val theme: String = "light",
    val isPublished: Boolean = false,
    val publicUrl: String? = null,
    val viewCount: Int = 0,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
)
